### PR TITLE
Fix for #1072

### PR DESCRIPTION
**Pull request by ThunderFrame**
Fix Crashes when there aren't any projects open

### CODE REVIEW

File: RetailCoder.VBE/UI/Command/MenuItems/ParserStateCommandBar.cs [hunk 1]
```diff
@@ -47,16 +47,18 @@ private void OnRefresh()
...
             _refreshButton = (CommandBarButton)commandbar.Controls.Add(MsoControlType.msoControlButton);
             ParentMenuItemBase.SetButtonImage(_refreshButton, Resources.arrow_circle_double, Resources.arrow_circle_double_mask);
             _refreshButton.Style = MsoButtonStyle.msoButtonIcon;
+            _refreshButton.Tag = "Refresh";
             _refreshButton.TooltipText = "Parse all opened projects";
             _refreshButton.Click += refreshButton_Click;
 
             _statusButton = (CommandBarButton)commandbar.Controls.Add(MsoControlType.msoControlButton);
             _statusButton.Style = MsoButtonStyle.msoButtonCaption;
+            _statusButton.Tag = "Status";
```
Comment (retailcoder) [inline RetailCoder.VBE/UI/Command/MenuItems/ParserStateCommandBar.cs]: What's the tag for?

File: RetailCoder.VBE/UI/Command/MenuItems/ParserStateCommandBar.cs [hunk 2]
```diff
@@ -47,16 +47,18 @@ private void OnRefresh()
 
         public void Initialize()
         {
-            var commandbar = _vbe.CommandBars.Add("Parsing", MsoBarPosition.msoBarTop, false, true);
+            var commandbar = _vbe.CommandBars.Add("Rubberduck", MsoBarPosition.msoBarTop, false, true);
```
Comment (retailcoder) [inline RetailCoder.VBE/UI/Command/MenuItems/ParserStateCommandBar.cs]: I'm completely ok with this, but then `ParserStateCommandBar` should be renamed `RubberduckCommandBar`.

File: Rubberduck.VBEEditor/Extensions/VbeExtensions.cs [hunk 3]
```diff
@@ -53,6 +54,36 @@ public static IHostApplication HostApplication(this VBE vbe)
...
+                        case "Microsoft Word":
+                            return new WordApp();
+                        case "Microsoft PowerPoint":
+                            return new PowerPointApp();
+                        case "Microsoft Outlook":
+                            return new OutlookApp();
+                        case "Microsoft Publisher":
+                            return new PublisherApp();
+                        case "AutoCAD":
+                            return null; //TODO - Confirm the button caption
```
Comment (retailcoder) [inline Rubberduck.VBEEditor/Extensions/VbeExtensions.cs]: Perhaps use `string.Contains` instead? why not `return new AutoCADApp();` anyway?

File: RetailCoder.VBE/UI/Command/MenuItems/ParserStateCommandBar.cs (hunk 1 above)
Comment (ThunderFrame) [inline RetailCoder.VBE/UI/Command/MenuItems/ParserStateCommandBar.cs]: Custom CommandBar controls don't let you assign an ID, and there isn't a Name property. Assigning a Tag, gives the user a way to find and interact with the control, as long as the tag is unique to the CommandBar.

File: Rubberduck.VBEEditor/Extensions/VbeExtensions.cs (hunk 3 above)
Comment (ThunderFrame) [inline Rubberduck.VBEEditor/Extensions/VbeExtensions.cs]: Contains seemed a bit flaky, for (future) things like Visio/Visio Viewer, Word and WordPerfect. And I don't know if AutoCAD has anything like AutoCAD Lite/Pro/Viewer.

I guess `case AutoCAD` *could* return a `new AutoCADApp()`, but it's unlikely that I've guessed the caption correctly. It felt safer to leave a placeholder, that in the worst case, doesn't return an App in the edge-case of there not being any projects present, rather then returning an `AutoCADApp` to something that might be expecting an AutoCAD Viewer.

As of this change, we've currently got 3 ways of determining the host Application, none of which is 100% reliable, all of the time. I think we ultimately need to try and get the host app from the processID of the VBE hWnd, as the preferred method, but this fix addresses the situation where VBE is running without any projects.

File: RetailCoder.VBE/UI/Command/MenuItems/ParserStateCommandBar.cs (hunk 2 above)
Comment (ThunderFrame) [inline RetailCoder.VBE/UI/Command/MenuItems/ParserStateCommandBar.cs]: I'll get this rename done later, once I'm off the beach. :p

Comment (retailcoder) [inline RetailCoder.VBE/UI/Command/MenuItems/ParserStateCommandBar.cs]: Enjoy the beach, I've renamed it already :sunglasses:

**Pull request merged**